include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -1386,6 +1400,124 @@ def ReplicateOp : MooreOp<"replicate", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Real Binary Operations
+//===----------------------------------------------------------------------===//
+
+class BinaryRealOpBase<string mnemonic, list<Trait> traits = []> :
+    MooreOp<mnemonic, traits # [Pure, SameOperandsAndResultType]> {
+  let arguments = (ins RealType:$lhs, RealType:$rhs);
+  let results = (outs RealType:$result);
+  let assemblyFormat = [{
+    $lhs `,` $rhs attr-dict `:` type($result)
+  }];
+}
+
+def AddRealOp : BinaryRealOpBase<"fadd"> {
+  let summary = "Addition";
+  let description = [{
+    Add the real operands.
+    See IEEE 1800-2017 § 11.4.3 "Arithmetic operators" and § 11.3.1 "Operators with real operands"
+  }];
+}
+
+def SubRealOp : BinaryRealOpBase<"fsub"> {
+  let summary = "Addition";
+  let description = [{
+    Subtract the RHS real operand from the LHS real operand.
+    See IEEE 1800-2017 § 11.4.3 "Arithmetic operators" and § 11.3.1 "Operators with real operands"
+  }];
+}
+
+def DivRealOp : BinaryRealOpBase<"fdiv"> {
+  let summary = "Addition";
+  let description = [{
+    Divide the LHS real operand by the LHS real operand.
+    See IEEE 1800-2017 § 11.4.3 "Arithmetic operators" and § 11.3.1 "Operators with real operands"
+  }];
+}
+
+def MulRealOp : BinaryRealOpBase<"fmul"> {
+  let summary = "Addition";
+  let description = [{
+    Multiply the RHS real operand with the LHS real operand.
+    See IEEE 1800-2017 § 11.4.3 "Arithmetic operators" and § 11.3.1 "Operators with real operands"
+  }];
+}
+
+def PowRealOp : BinaryRealOpBase<"fpow"> {
+  let summary = "Addition";
+  let description = [{
+    Exponentiate the LHS real base with the RHS real exponent.
+    See IEEE 1800-2017 § 11.4.3 "Arithmetic operators" and § 11.3.1 "Operators with real operands"
+  }];
+}
+
+class LogicalEqRealOpBase<string mnemonic> : MooreOp<mnemonic, [
+  Pure,
+  Commutative,
+  SameTypeOperands,
+  ResultIsSingleBitMatchingInputDomain<"result", "lhs">
+]> {
+  let description = [{
+    Compares the bits in the left- and right-hand side operand and returns a
+    single bit 0, 1, or X result. If all corresponding bits in the left- and
+    right-hand side are equal, and all are 0 or 1 (not X or Z), the two operands
+    are considered equal (`eq` returns 1, `ne` returns 0). If any bits are not
+    equal, but all are 0 or 1, the two operands are considered not equal (`eq`
+    returns 0, `ne` returns 1). If any bit in the two operands is Z or X,
+    returns X. `eq` corresponds to the `==` operator and `ne` to the `!=`
+    operator.
+
+    See IEEE 1800-2017 § 11.4.5 "Equality operators" and § 11.3.1 "Operators with real operands".
```

Comment:
Are there any X values in real comparisons? I thought these were limited to four-state integer comparisons, and comparison ops on reals would just produce a boolean.

Reply:
This was left over from essentially copying 😅